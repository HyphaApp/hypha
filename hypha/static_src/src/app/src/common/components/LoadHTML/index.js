import * as React from "react";
import PropTypes from 'prop-types';


const LoadHTML = props => {
  return <div className="form__group rich-text">
    <div dangerouslySetInnerHTML={{ __html: props.text }} />
  </div>
}

LoadHTML.propTypes={
  text: PropTypes.string
}

LoadHTML.displayName = 'LoadHTML';
export default LoadHTML;
