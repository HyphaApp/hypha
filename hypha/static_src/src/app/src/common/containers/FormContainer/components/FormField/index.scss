
.error {
    color: #ec2d0b;
    font-size:12px;
    line-height: 14px;
    min-height: 15px;
}
