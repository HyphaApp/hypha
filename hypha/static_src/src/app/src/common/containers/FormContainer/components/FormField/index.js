import * as React from "react";
import  "./index.scss";
import TextBox from "@common/components/TextBox";
import DropDown from "@common/components/DropDown";
import Radio from "@common/components/Radio";
import TinyMCE from "@common/components/TinyMCE";
import ScoredAnswerWidget from "@common/components/ScoredAnswerWidget";
import LoadHTML from "@common/components/LoadHTML";
import Textarea from "@common/components/Textarea";
import CheckBox from "@common/components/CheckBox";
import PropTypes from 'prop-types';


import _ from "lodash";

class FormField extends React.Component {
  static propTypes = {
    onChange: PropTypes.func,
    widget: PropTypes.object,
    kwargs: PropTypes.object,
    value: PropTypes.node,
    error: PropTypes.string,
    fieldProps: PropTypes.object,
    type: PropTypes.string,

  }

    onChange = (name, value) => {
      this.props.onChange(name, value)
    }

    getType() {
      return this.props.widget && this.props.widget.type ? this.props.widget.type : this.props.type;
    }

    getHelperprops() {
      return {
        text: this.props.kwargs.help_text,
        link: this.props.kwargs.help_link,
      }
    }

    renderField() {
        const {value, kwargs, widget,  ...fieldProps} = this.props;
      switch (this.getType()) {
          case "EmailInput":
          case "URLInput":
          case "TextInput":
            return <TextBox 
              label={kwargs.label}
              required={kwargs.required}
              name={fieldProps.name}
              value={value}
              onChange={this.onChange}
              id={fieldProps.name}
              helperProps={this.getHelperprops()}
            />;

          case "TinyMCE":
            return <TinyMCE 
              label={kwargs.label}
              name={fieldProps.name}
              onChange={this.onChange}
              value={value}
              id={fieldProps.name}
              init={widget.mce_attrs}
              required={kwargs.required}
              helperProps={this.getHelperprops()}
              
            />;

        case "Textarea":
          return <Textarea
            label={kwargs.label}
            name={fieldProps.name}
            onChange={this.onChange}
            value={value}
            id={fieldProps.name}
            widget={widget}
            required={kwargs.required}
            helperProps={this.getHelperprops()}
            
          />;
            
        case "Select":
          return <DropDown 
            label={kwargs.label}
            name={fieldProps.name}
            value={value}
            onChange={this.onChange}
            id={fieldProps.name}
            choices={kwargs.choices}
            required={kwargs.required}
            helperProps={this.getHelperprops()}
            
          />;

        case "RadioSelect":
          return <Radio 
            value={value}
            help_text={kwargs.help_text}
            label={kwargs.label}
            choices={kwargs.choices}
            name={fieldProps.name}
            onChange={this.onChange}
            required={kwargs.required}
            helperProps={this.getHelperprops()}
            
          />;

        case "CheckboxInput":
          return <CheckBox
            value={value}
            help_text={kwargs.help_text}
            label={kwargs.label}
            choices={kwargs.choices}
            name={fieldProps.name}
            onChange={this.onChange}
            required={kwargs.required}
            helperProps={this.getHelperprops()}
            
          />;
          
        case "ScoredAnswerWidget":
          return <ScoredAnswerWidget
            value={_.isArray(value) ? value.asMutable() : []}
            help_text={kwargs.help_text}
            label={kwargs.label}
            name={fieldProps.name}
            onChange={this.onChange}
            kwargs={kwargs}
            widget={widget.widgets}
            required={kwargs.required}
            helperProps={this.getHelperprops()}
            
          />;

        case "LoadHTML":
          return <LoadHTML 
            text={kwargs.text}
          />;
      
        default:
          return <div>Unknown field type {this.getType()}</div>
        }
    }
        
    render() {
        return <div className="formFieldcontainer" > 
            {this.renderField()}
            <div className="error">
                {this.props.error}
            </div>
        </div>
    }
}

    
export default FormField;
    
