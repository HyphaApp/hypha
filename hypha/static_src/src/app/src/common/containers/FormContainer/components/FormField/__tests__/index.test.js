import React from "react";
import { mount, shallow } from "enzyme";
import sinon from "sinon";
import FormField from "../index";
import * as enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
enzyme.configure({ adapter: new Adapter() });
import TextBox from "@common/components/TextBox";
import DropDown from "@common/components/DropDown";
import Radio from "@common/components/Radio";
import TinyMCE from "@common/components/TinyMCE";
import ScoredAnswerWidget from "@common/components/ScoredAnswerWidget";
import LoadHTML from "@common/components/LoadHTML";
import Textarea from "@common/components/Textarea";
import CheckBox from "@common/components/CheckBox";
import PageDownWidget from "@common/components/PageDownWidget";
import PropTypes from 'prop-types';


describe("Test form field component should render TextBox", () => {
    const fieldProps = {name : 'name text'}
    const value = ["value"]
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "EmailInput", mce_attrs: "mce_attrs1", widgets: []}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1"}
    const type = "type2"

    const subject = mount(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and TextBox component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      subject.find('TextBox').props().onChange();
      expect(onChange).toHaveBeenCalled();
      expect(subject.containsMatchingElement(<TextBox />)).toEqual(true);
    });
  
    test('render a form field component with textbox', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

describe("Test form field component should render TinyMCE", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "", mce_attrs: {key : "mce_attrs1"}, widgets: []}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1"}
    const type = "TinyMCE"

    const subject = shallow(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and TinyMCE component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      subject.find('TinyMCE').props().onChange('name text', ["value"]);
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith('name text', ["value"]);
      expect(subject.containsMatchingElement(<TinyMCE />)).toEqual(true);
    });
  
    test('render a form field component with TinyMCE', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

describe("Test form field component should render Textarea", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "Textarea", mce_attrs: {key : "mce_attrs1"}, widgets: [], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1"}
    const type = "Textarea"

    const subject = mount(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and Textarea component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      subject.find('Textarea').props().onChange('name text', ["value"]);
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith('name text', ["value"]);
      expect(subject.containsMatchingElement(<Textarea />)).toEqual(true);
    });
  
    test('render a form field component with Textarea', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test form field component should render DropDown", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "", mce_attrs: {key : "mce_attrs1"}, widgets: [], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1"}
    const type = "Select"

    const subject = mount(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and DropDown component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      subject.find('DropDown').props().onChange('name text', ["value"]);
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith('name text', ["value"]);
      expect(subject.containsMatchingElement(<DropDown />)).toEqual(true);
    });
  
    test('render a form field component with DropDown', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test form field component should render Radio", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "RadioSelect", mce_attrs: {key : "mce_attrs1"}, widgets: [], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1"}
    const type = ""

    const subject = mount(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and Radio component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      subject.find('Radio').props().onChange('name text', ["value"]);
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith('name text', ["value"]);
      expect(subject.containsMatchingElement(<Radio />)).toEqual(true);
    });
  
    test('render a form field component with Radio', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test form field component should render CheckBox", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "", mce_attrs: {key : "mce_attrs1"}, widgets: [], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1"}
    const type = "CheckboxInput"

    const subject = mount(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and CheckBox component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      subject.find('CheckBox').props().onChange('name text', ["value"]);
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith('name text', ["value"]);
      expect(subject.containsMatchingElement(<CheckBox />)).toEqual(true);
    });
  
    test('render a form field component with CheckBox', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test form field component should render ScoredAnswerWidget", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "", mce_attrs: {key : "mce_attrs1"}, widgets: [{mce_attrs: {key : "mce_attrs2"}}], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1", fields: [{},{choices: [1]}]}
    const type = "ScoredAnswerWidget"

    const subject = shallow(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and ScoredAnswerWidget component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      subject.find('ScoredAnswerWidget').props().onChange('name text', ["value"]);
      expect(subject.find('ScoredAnswerWidget').props().value).toEqual([])
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith('name text', ["value"]);
      expect(subject.containsMatchingElement(<ScoredAnswerWidget />)).toEqual(true);
    });
  
    test('render a form field component with ScoredAnswerWidget', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test form field component should render LoadHTML", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "", mce_attrs: {key : "mce_attrs1"}, widgets: [{mce_attrs: {key : "mce_attrs2"}}], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1", fields: [{},{choices: [1]}]}
    const type = "LoadHTML"

    const subject = mount(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and LoadHTML component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      expect(subject.containsMatchingElement(<LoadHTML />)).toEqual(true);
    });
  
    test('render a form field component with LoadHTML', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test form field component should render PagedownWidget", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "PagedownWidget", mce_attrs: {key : "mce_attrs1"}, widgets: [{mce_attrs: {key : "mce_attrs2"}}], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1", fields: [{},{choices: [1]}]}
    const type = ""

    const subject = shallow(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes and PagedownWidget component", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      expect(subject.containsMatchingElement(<PageDownWidget />)).toEqual(true);
      subject.find('PageDownWidget').props().onChange('name text', ["value"]);
      expect(onChange).toHaveBeenCalled();
      expect(onChange).toHaveBeenCalledWith('name text', ["value"]);
    });
  
    test('render a form field component with PagedownWidget', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });

  describe("Test form field component should render unknown type", () => {
    const fieldProps = {name : 'name text'}
    const value = "value"
    const error = "error text"
    const onChange = jest.fn()
    const widget = {type: "none type", mce_attrs: {key : "mce_attrs1"}, widgets: [{mce_attrs: {key : "mce_attrs2"}}], attrs : { cols: 1, rows: 1}}
    const kwargs = { help_text: "help_text", help_link: "link1", label: "label1", required: true, choices : [[0,1]], text: "text1", fields: [{},{choices: [1]}]}
    const type = "PageDownWidget"

    const subject = mount(<FormField
      fieldProps={fieldProps}
      onChange={onChange}
      value={value}
      error={error}
      widget={widget}
      kwargs={kwargs}
      type={type}
    />);
  
    it("Shoud render without issues", () => {
      expect(subject.length).toBe(1);
    });
  
    it("Should have searched classes", () => {
      expect(subject.find('.formFieldcontainer').length).toBe(1);
      expect(subject.find('.error').text()).toBe(error);
      expect(subject.find('.formFieldcontainer').first().text().includes("Unknown field type none type")).toBe(true);
      expect(subject.containsMatchingElement(<PageDownWidget />)).toEqual(false);
    });
  
    test('render a form field component with unknown type passed', () => {
      expect(subject).toMatchSnapshot();
    });
  
  });
