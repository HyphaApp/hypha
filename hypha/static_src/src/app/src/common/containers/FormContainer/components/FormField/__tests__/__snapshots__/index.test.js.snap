// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test form field component should render CheckBox render a form field component with CheckBox 1`] = `
<FormField
  error="error text"
  fieldProps={
    Object {
      "name": "name text",
    }
  }
  kwargs={
    Object {
      "choices": Array [
        Array [
          0,
          1,
        ],
      ],
      "help_link": "link1",
      "help_text": "help_text",
      "label": "label1",
      "required": true,
      "text": "text1",
    }
  }
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "name text",
          Array [
            "value",
          ],
        ],
      ],
    }
  }
  type="CheckboxInput"
  value="value"
  widget={
    Object {
      "attrs": Object {
        "cols": 1,
        "rows": 1,
      },
      "mce_attrs": Object {
        "key": "mce_attrs1",
      },
      "type": "",
      "widgets": Array [],
    }
  }
>
  <div
    className="formFieldcontainer"
  >
    <CheckBox
      choices={
        Array [
          Array [
            0,
            1,
          ],
        ]
      }
      help_text="help_text"
      helperProps={
        Object {
          "link": "link1",
          "text": "help_text",
        }
      }
      label="label1"
      onChange={[Function]}
      required={true}
      value="value"
    >
      <div
        className="form__group  form__group--checkbox form__group--wrap"
      >
        <label
          className="form__question form__question--boolean_field checkbox_input"
        >
          <span>
            label1
          </span>
          <span
            className="form__required"
          >
             *
          </span>
        </label>
        <div
          className="form__item"
          onClick={[Function]}
        >
          <input
            checked="checked"
            onChange={[Function]}
            type="checkbox"
          />
          <label />
        </div>
        <hr />
        <HelperComponent
          link="link1"
          text="help_text"
        >
          <p
            className="form__help"
            dangerouslySetInnerHTML={
              Object {
                "__html": "help_text",
              }
            }
          />
          <p
            className="form__help-link"
          >
            <a
              href="link1"
              rel="noopener noreferrer"
              target="_blank"
            >
              See help guide for more information.
              <svg
                className="form__open-icon"
              >
                <use
                  href="#open-in-new-tab"
                />
              </svg>
            </a>
          </p>
        </HelperComponent>
      </div>
    </CheckBox>
    <div
      className="error"
    >
      error text
    </div>
  </div>
</FormField>
`;

exports[`Test form field component should render DropDown render a form field component with DropDown 1`] = `
<FormField
  error="error text"
  fieldProps={
    Object {
      "name": "name text",
    }
  }
  kwargs={
    Object {
      "choices": Array [
        Array [
          0,
          1,
        ],
      ],
      "help_link": "link1",
      "help_text": "help_text",
      "label": "label1",
      "required": true,
      "text": "text1",
    }
  }
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "name text",
          Array [
            "value",
          ],
        ],
      ],
    }
  }
  type="Select"
  value="value"
  widget={
    Object {
      "attrs": Object {
        "cols": 1,
        "rows": 1,
      },
      "mce_attrs": Object {
        "key": "mce_attrs1",
      },
      "type": "",
      "widgets": Array [],
    }
  }
>
  <div
    className="formFieldcontainer"
  >
    <DropDown
      choices={
        Array [
          Array [
            0,
            1,
          ],
        ]
      }
      helperProps={
        Object {
          "link": "link1",
          "text": "help_text",
        }
      }
      label="label1"
      onChange={[Function]}
      required={true}
      value="value"
    >
      <div
        className="form__group "
      >
        <label
          className="form__question form__question--choice_field select"
          required=""
        >
          <span>
            label1
          </span>
          <span
            className="form__required"
          >
             *
          </span>
        </label>
        <HelperComponent
          link="link1"
          text="help_text"
        >
          <p
            className="form__help"
            dangerouslySetInnerHTML={
              Object {
                "__html": "help_text",
              }
            }
          />
          <p
            className="form__help-link"
          >
            <a
              href="link1"
              rel="noopener noreferrer"
              target="_blank"
            >
              See help guide for more information.
              <svg
                className="form__open-icon"
              >
                <use
                  href="#open-in-new-tab"
                />
              </svg>
            </a>
          </p>
        </HelperComponent>
        <div
          className="form__item"
        >
          <div
            className="form__select"
          >
            <select
              onChange={[Function]}
              value="value"
            >
              <option
                key="0"
                value={0}
              >
                1
              </option>
            </select>
          </div>
        </div>
      </div>
    </DropDown>
    <div
      className="error"
    >
      error text
    </div>
  </div>
</FormField>
`;

exports[`Test form field component should render LoadHTML render a form field component with LoadHTML 1`] = `
<FormField
  error="error text"
  fieldProps={
    Object {
      "name": "name text",
    }
  }
  kwargs={
    Object {
      "choices": Array [
        Array [
          0,
          1,
        ],
      ],
      "fields": Array [
        Object {},
        Object {
          "choices": Array [
            1,
          ],
        },
      ],
      "help_link": "link1",
      "help_text": "help_text",
      "label": "label1",
      "required": true,
      "text": "text1",
    }
  }
  onChange={[MockFunction]}
  type="LoadHTML"
  value="value"
  widget={
    Object {
      "attrs": Object {
        "cols": 1,
        "rows": 1,
      },
      "mce_attrs": Object {
        "key": "mce_attrs1",
      },
      "type": "",
      "widgets": Array [
        Object {
          "mce_attrs": Object {
            "key": "mce_attrs2",
          },
        },
      ],
    }
  }
>
  <div
    className="formFieldcontainer"
  >
    <LoadHTML
      text="text1"
    >
      <div
        className="form__group rich-text"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "text1",
            }
          }
        />
      </div>
    </LoadHTML>
    <div
      className="error"
    >
      error text
    </div>
  </div>
</FormField>
`;

exports[`Test form field component should render PagedownWidget render a form field component with PagedownWidget 1`] = `
<div
  className="formFieldcontainer"
>
  <PageDownWidget
    helperProps={
      Object {
        "link": "link1",
        "text": "help_text",
      }
    }
    init={
      Object {
        "key": "mce_attrs1",
      }
    }
    label="label1"
    onChange={[Function]}
    required={true}
    value="value"
  />
  <div
    className="error"
  >
    error text
  </div>
</div>
`;

exports[`Test form field component should render Radio render a form field component with Radio 1`] = `
<FormField
  error="error text"
  fieldProps={
    Object {
      "name": "name text",
    }
  }
  kwargs={
    Object {
      "choices": Array [
        Array [
          0,
          1,
        ],
      ],
      "help_link": "link1",
      "help_text": "help_text",
      "label": "label1",
      "required": true,
      "text": "text1",
    }
  }
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "name text",
          Array [
            "value",
          ],
        ],
      ],
    }
  }
  type=""
  value="value"
  widget={
    Object {
      "attrs": Object {
        "cols": 1,
        "rows": 1,
      },
      "mce_attrs": Object {
        "key": "mce_attrs1",
      },
      "type": "RadioSelect",
      "widgets": Array [],
    }
  }
>
  <div
    className="formFieldcontainer"
  >
    <Radio
      choices={
        Array [
          Array [
            0,
            1,
          ],
        ]
      }
      help_text="help_text"
      helperProps={
        Object {
          "link": "link1",
          "text": "help_text",
        }
      }
      label="label1"
      onChange={[Function]}
      required={true}
      value="value"
    >
      <div
        className="form__group "
      >
        <label
          className="form__question form__question--choice_field radio_select"
          required=""
        >
          <span>
            label1
          </span>
          <span
            className="form__required"
          >
             *
          </span>
        </label>
        <HelperComponent
          link="link1"
          text="help_text"
        >
          <p
            className="form__help"
            dangerouslySetInnerHTML={
              Object {
                "__html": "help_text",
              }
            }
          />
          <p
            className="form__help-link"
          >
            <a
              href="link1"
              rel="noopener noreferrer"
              target="_blank"
            >
              See help guide for more information.
              <svg
                className="form__open-icon"
              >
                <use
                  href="#open-in-new-tab"
                />
              </svg>
            </a>
          </p>
        </HelperComponent>
        <div
          className="form__item"
        >
          <ul
            className="grid grid--no-margin grid--two"
          >
            <li
              className="form__item"
              key="0"
              onClick={[Function]}
            >
              <input
                checked=""
                onChange={[Function]}
                type="radio"
                value={0}
              />
              <label
                className="form__label"
                htmlFor="undefined-0"
              >
                1
              </label>
            </li>
          </ul>
        </div>
      </div>
    </Radio>
    <div
      className="error"
    >
      error text
    </div>
  </div>
</FormField>
`;

exports[`Test form field component should render ScoredAnswerWidget render a form field component with ScoredAnswerWidget 1`] = `
<div
  className="formFieldcontainer"
>
  <ScoredAnswerWidget
    help_text="help_text"
    helperProps={
      Object {
        "link": "link1",
        "text": "help_text",
      }
    }
    kwargs={
      Object {
        "choices": Array [
          Array [
            0,
            1,
          ],
        ],
        "fields": Array [
          Object {},
          Object {
            "choices": Array [
              1,
            ],
          },
        ],
        "help_link": "link1",
        "help_text": "help_text",
        "label": "label1",
        "required": true,
        "text": "text1",
      }
    }
    label="label1"
    onChange={[Function]}
    required={true}
    value={Array []}
    widget={
      Array [
        Object {
          "mce_attrs": Object {
            "key": "mce_attrs2",
          },
        },
      ]
    }
  />
  <div
    className="error"
  >
    error text
  </div>
</div>
`;

exports[`Test form field component should render TextBox render a form field component with textbox 1`] = `
<FormField
  error="error text"
  fieldProps={
    Object {
      "name": "name text",
    }
  }
  kwargs={
    Object {
      "choices": Array [
        Array [
          0,
          1,
        ],
      ],
      "help_link": "link1",
      "help_text": "help_text",
      "label": "label1",
      "required": true,
      "text": "text1",
    }
  }
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          undefined,
          undefined,
        ],
      ],
    }
  }
  type="type2"
  value={
    Array [
      "value",
    ]
  }
  widget={
    Object {
      "mce_attrs": "mce_attrs1",
      "type": "EmailInput",
      "widgets": Array [],
    }
  }
>
  <div
    className="formFieldcontainer"
  >
    <TextBox
      helperProps={
        Object {
          "link": "link1",
          "text": "help_text",
        }
      }
      label="label1"
      onChange={[Function]}
      required={true}
      value={
        Array [
          "value",
        ]
      }
    >
      <div
        className="form__group "
      >
        <label
          className="form__question form__question--char_field text_input"
        >
          <span>
            label1
          </span>
          <span
            className="form__required"
          >
             *
          </span>
        </label>
        <HelperComponent
          link="link1"
          text="help_text"
        >
          <p
            className="form__help"
            dangerouslySetInnerHTML={
              Object {
                "__html": "help_text",
              }
            }
          />
          <p
            className="form__help-link"
          >
            <a
              href="link1"
              rel="noopener noreferrer"
              target="_blank"
            >
              See help guide for more information.
              <svg
                className="form__open-icon"
              >
                <use
                  href="#open-in-new-tab"
                />
              </svg>
            </a>
          </p>
        </HelperComponent>
        <div
          className="form__item"
        >
          <input
            onChange={[Function]}
            type="text"
            value={
              Array [
                "value",
              ]
            }
          />
        </div>
      </div>
    </TextBox>
    <div
      className="error"
    >
      error text
    </div>
  </div>
</FormField>
`;

exports[`Test form field component should render Textarea render a form field component with Textarea 1`] = `
<FormField
  error="error text"
  fieldProps={
    Object {
      "name": "name text",
    }
  }
  kwargs={
    Object {
      "choices": Array [
        Array [
          0,
          1,
        ],
      ],
      "help_link": "link1",
      "help_text": "help_text",
      "label": "label1",
      "required": true,
      "text": "text1",
    }
  }
  onChange={
    [MockFunction] {
      "calls": Array [
        Array [
          "name text",
          Array [
            "value",
          ],
        ],
      ],
    }
  }
  type="Textarea"
  value="value"
  widget={
    Object {
      "attrs": Object {
        "cols": 1,
        "rows": 1,
      },
      "mce_attrs": Object {
        "key": "mce_attrs1",
      },
      "type": "Textarea",
      "widgets": Array [],
    }
  }
>
  <div
    className="formFieldcontainer"
  >
    <Textarea
      helperProps={
        Object {
          "link": "link1",
          "text": "help_text",
        }
      }
      label="label1"
      onChange={[Function]}
      required={true}
      value="value"
      widget={
        Object {
          "attrs": Object {
            "cols": 1,
            "rows": 1,
          },
          "mce_attrs": Object {
            "key": "mce_attrs1",
          },
          "type": "Textarea",
          "widgets": Array [],
        }
      }
    >
      <div
        className="form__group "
      >
        <div
          className="form__group  form__group--wrap"
          data-word-limit="1000"
        >
          <label
            className="form__question form__question--char_field textarea"
          >
            <span>
              label1
            </span>
            <span
              className="form__required"
            >
               *
            </span>
          </label>
          <HelperComponent
            link="link1"
            text="help_text"
          >
            <p
              className="form__help"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "help_text",
                }
              }
            />
            <p
              className="form__help-link"
            >
              <a
                href="link1"
                rel="noopener noreferrer"
                target="_blank"
              >
                See help guide for more information.
                <svg
                  className="form__open-icon"
                >
                  <use
                    href="#open-in-new-tab"
                  />
                </svg>
              </a>
            </p>
          </HelperComponent>
          <div
            className="form__item"
          >
            <textarea
              cols={1}
              onChange={[Function]}
              rows={1}
              value="value"
            />
          </div>
        </div>
      </div>
    </Textarea>
    <div
      className="error"
    >
      error text
    </div>
  </div>
</FormField>
`;

exports[`Test form field component should render TinyMCE render a form field component with TinyMCE 1`] = `
<div
  className="formFieldcontainer"
>
  <TinyMCE
    helperProps={
      Object {
        "link": "link1",
        "text": "help_text",
      }
    }
    init={
      Object {
        "key": "mce_attrs1",
      }
    }
    label="label1"
    onChange={[Function]}
    required={true}
    value="value"
  />
  <div
    className="error"
  >
    error text
  </div>
</div>
`;

exports[`Test form field component should render unknown type render a form field component with unknown type passed 1`] = `
<FormField
  error="error text"
  fieldProps={
    Object {
      "name": "name text",
    }
  }
  kwargs={
    Object {
      "choices": Array [
        Array [
          0,
          1,
        ],
      ],
      "fields": Array [
        Object {},
        Object {
          "choices": Array [
            1,
          ],
        },
      ],
      "help_link": "link1",
      "help_text": "help_text",
      "label": "label1",
      "required": true,
      "text": "text1",
    }
  }
  onChange={[MockFunction]}
  type="PageDownWidget"
  value="value"
  widget={
    Object {
      "attrs": Object {
        "cols": 1,
        "rows": 1,
      },
      "mce_attrs": Object {
        "key": "mce_attrs1",
      },
      "type": "none type",
      "widgets": Array [
        Object {
          "mce_attrs": Object {
            "key": "mce_attrs2",
          },
        },
      ],
    }
  }
>
  <div
    className="formFieldcontainer"
  >
    <div>
      Unknown field type 
      none type
    </div>
    <div
      className="error"
    >
      error text
    </div>
  </div>
</FormField>
`;
